lnp2p/src/bolt/bolt2.rs

```diff
@@ -93,6 +94,34 @@ impl ChannelType {
     }
 }
 
+impl TryFrom<FlagVec> for ChannelType {
+    type Error = lightning_encoding::Error;
+
+    fn try_from(mut flags: FlagVec) -> Result<Self, Self::Error> {
+        if flags.shrink() {
+            return Err(lightning_encoding::Error::DataIntegrityError(s!(
+                "non-minimal channel type encoding"
+            )));
+        } else if flags.as_inner() == &[] as &[u8] {
```

Comment:
I think this doesn't seem right because if channel-type was not defined, FlagVec::lightning_deserialize (see [here](https://github.com/LNP-WG/lnp-core/blob/2d7c0f66a46cc939f1b9057bfc582fdaf9cef2cd/lnp2p/src/bolt/bolt2.rs#L187)) returns FlagVec::default() (vec![0u8; 1])  instead of an empty vector.

```suggestion
        } else if flags == FlagVec::new() {
```